spec/take_away_ui_spec.rb

```diff
@@ -0,0 +1,215 @@
+  context 'Submiting orders by text or selecting and adding to basket individually' do
+        let(:test_order) {"2 carbonara, 1 tiramisu and I dunno"}
+        let(:selection1) {["Cake",3]}
+        let(:selection2) {["Tiramisu", 1]}
+
+
+        before do
+          allow(STDOUT).to receive(:p).with("What would you like to eat? Eg '2 Carbonara and a Tiramisu'")
+          allow(STDIN).to receive(:gets).and_return(test_order)
+          allow(dummy_order_klass).to receive(:new).with(test_order, default_menu_dinner_mains).and_return(dummy_order)
+          allow(dummy_basket_klass).to receive(:new).with([selection1], italian_menu).and_return(dummy_basket)
+        end
+
+        describe '#take_order' do
+
+          before do
+            allow(dummy_order).to receive(:add_to_basket)
+          end
+
+
+          it 'instantiates a new order' do
+            expect(dummy_order_klass).to receive(:new).and_return(dummy_order)
+            menu.take_order
+          end
+
+          it 'adds the order to basket' do
+            expect(dummy_order).to receive(:add_to_basket)
+            menu.take_order
+          end
+        end
+
+      describe '#select_dish' do
+          it {is_expected.to respond_to(:select_dish).with(2).arguments}
+
+
+          before do
+            allow(dummy_basket).to receive(:selection).and_return([[selection1]])
+            allow(dummy_basket_klass).to receive(:new).with(menu.selection, default_menu_dinner_mains).and_return(dummy_basket)
+            menu.select_dish("Cake", 3)
+          end
+
+          it 'puts dish selections into selection1 array' do
+            expect(menu.selection).to include selection1
+          end
+
+          it 'instantiates a new basket on first selection' do
+            expect(menu.basket).to eq dummy_basket
+          end
+
+          it 'puts dish selections into selection array' do
+            menu.select_dish("Tiramisu", 1)
+            expect(menu.basket.selection).to include selection2
+          end
+
+          it 'does not instantiate a new basket on second selections' do
+            menu.select_dish("Tiramisu", 1)
+            expect(menu.basket).to eq dummy_basket
+          end
+
+          it 'selecting dishes increases the size of selection array' do
+            expect {menu.select_dish("Tiramisu", 1)}.to change{menu.basket.selection.size}.by(1)
+          end
+
+
+
+      describe "#review_order" do
+            let(:sample_bill) {double :sample_bill}
+            let(:sample_total) {double :sample_total}
+
+            before do
+              allow(STDOUT).to receive(:puts).and_return("")
+              allow(dummy_basket).to receive(:itemised_bill).and_return sample_bill
+              allow(sample_bill).to receive(:each)
+              allow(dummy_basket).to receive(:total_bill).and_return sample_total
+            end
+
+            it 'will retrieve the itemised_bill' do
+              expect(dummy_basket).to receive(:itemised_bill).and_return sample_bill
```

Comment:
Metrics/LineLength: Line is too long. [84/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)